src/test/clojure/core2/core/datalog_test.clj

```diff
@@ -938,6 +939,267 @@
                            (assoc :basis {:tx !tx}))))
               "b is unified")))))
 
+(deftest test-basic-rules
+  (t/is (= '[[:triple {:e [:logic-var i], :a :age, :v [:logic-var age]}]
+             [:rule {:name over-twenty-one?, :args [[:logic-var age] [:literal 21]]}]]
+           (s/conform :core2.core.datalog/where '[[i :age age]
+                                                  (over-twenty-one? age 21)])))
+
+  (t/is (= '[{:head {:name over-twenty-one?, :args [age]},
+              :body
+              [[:call
+                {:form [:fn-call {:f >=, :args [[:logic-var age] [:literal 21]]}]}]]}
+             {:head {:name over-twenty-one?, :args [age]},
+              :body
+              [[:anti-join
+                {:not-exists not-exists?,
+                 :args [age],
+                 :terms
+                 [[:call
+                   {:form
+                    [:fn-call {:f <, :args [[:logic-var age] [:literal 21]]}]}]]}]]}]
+
+           (s/conform :core2.core.datalog/rules '[[(over-twenty-one? age)
+                                                   [(>= age 21)]]
+                                                  [(over-twenty-one? age)
+                                                   (not-exists? [age]
+                                                                [(< age 21)])]])))
+
+  (let [tx (c2/submit-tx tu/*node* [[:put {:id :ivan :name "Ivan" :last-name "Ivanov" :age 21}]
+                                    [:put {:id :petr :name "Petr" :last-name "Petrov" :age 18}]
+                                    [:put {:id :georgy :name "Georgy" :last-name "George" :age 17}]])
+        q (fn q [query & args]
+            (apply c2/q tu/*node*
+                   (-> query
+                       (assoc :basis {:tx tx}))
```

Comment:
shouldn't need this `assoc :basis` now, unless you're querying at different tx-times